src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,147 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> node;
+        if (size == 0) {
+            node = new Node<>(null, value, null);
+            head = node;
+            tail = node;
+            size++;
+            return;
+        }
+        if (index == 0) {
+            Node<T> nodeSecond = head;
+            node = new Node<>(null,value, nodeSecond);
+            nodeSecond.prev = node;
+            head = node;
+            size++;
+            return;
+        }
+        Node<T> firstNode = nodeForIndex(index - 1);
+        Node<T> secondNode = null;
+        if (size > index) {
+            secondNode = nodeForIndex(index);
+        }
+        Node<T> addedNode = new Node<>(firstNode, value, secondNode);
+        firstNode.next = addedNode;
+        if (size == index) {
+            tail = addedNode;
+        }
+        size++;
```

Comment:
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        Node<T> next = nodeForIndex(index);
        Node<T> prev = next.prev;
        Node<T> newNode = new Node<>(prev, value, next);
        if (prev == null) {
            first = newNode;
        } else {
            prev.next = newNode;
        }
        next.prev = newNode;
        size++;
```

we can simplify this